### PR TITLE
use an IdentitySet so as to avoid calling hashCode() when reference a…

### PR DESCRIPTION
**Pull request by JerryShea**
…ccounting https://github.com/ChronicleEnterprise/Chronicle-FIX/issues/838

### CODE REVIEW

**Review by nicktindall — APPROVED** [review]
Nice one!

**Comment by JerryShea** [comment]
Everyone else please feel to add yourselves as reviewers

File: src/main/java/net/openhft/chronicle/core/io/AbstractCloseable.java [hunk 1]
```diff
@@ -159,7 +160,7 @@ public static void assertCloseablesClosed() {
 
         synchronized (traceSet) {
             traceSet.removeIf(o -> o == null || o.isClosing());
-            Set<Closeable> nested = new HashSet<>();
+            Set<Closeable> nested = Collections.newSetFromMap(new IdentityHashMap<>());
             for (Closeable key : traceSet) {
                 addNested(nested, key, 1);
             }
```
Comment (nicktindall) [inline src/main/java/net/openhft/chronicle/core/io/AbstractCloseable.java:166]: I think the HashSet on the following line also needs to be an identity set

Comment (JerryShea) [reply src/main/java/net/openhft/chronicle/core/io/AbstractCloseable.java:166]: good call - thanks

**Comment by minborg** [comment]
This was a very good catch! Well done!

**Review by alamar — APPROVED** [review]

**Pull request merged**